source/game.py

```diff
@@ -1,22 +1,101 @@
+import string
+from os import path
+
+from blessed import Terminal
+from blessed.keyboard import Keystroke
+from highscore import Highscore
+from images import Image
+from menus.utils import Menu, State
 from sliding_puzzle import Puzzle
 
 
-class Game:
-    """Game class"""
+class Game(Menu):
+    """Game menu where the user is playing."""
 
     def __init__(self, image: str, difficulty: int = 1):
         """Init function
 
-        Args:
-            image (str): image path
-            difficulty (int, optional): integer ranging from 1 to 5. Defaults to 1.
+        :param image: Image path
+        :param difficulty: integer ranging from 1 to 5. Defaults to 1.
         """
+        self.selected = 0
+        self.player_name = ''
+        self.image_name = path.basename(image).split('.')[0]
+
         difficulty = min(5, max(difficulty, 1))
-        # Convert image to ASCII
-        # self.ascii = toAscii(image)
-        with open('../resources/ascii_art/amogus.txt', 'r') as f:
-            self.ascii = f.read()
-        # Need to adjust the dimensions for sliding_puzzle.Puzzle
-        vertical = 0
-        horizontal = 0
-        self.puzzle = Puzzle(self.ascii, vertical, horizontal)
+
+        self.image = Image(image, 90)
+
+        self.puzzle = Puzzle(self.image.generate_ascii, 3, 3)
+        self.puzzle.shuffle()
+
+    def render(self, term: Terminal) -> str:
```

Comment:
```suggestion
    def render(self, term: Interface) -> str:
```
We use Interface instead of Terminal